### PR TITLE
45 register page

### LINKED ISSUE #45: Create RegisterPage component  with basic functionality
Should implement design from Figma #31 (This issue can only be closed if the page looks like a project)
Should contain:
- Input for login
- Input for email
- Input for username
- Image input for user avatar
- Input for password and confirm password
- Link to login page
Should implement basic functionality (form validation, display errors with #54 ) which can be used later to send data via http
Should be added to router.
Inputs should be done with #84

### PR DESCRIPTION
**Pull request by karolkozlo**
closes #45

### CODE REVIEW

File: frontend/src/pages/RegisterPage.vue [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+      }
+      return false;
+    },
+    createAccount() {
+      if(this.formValidation()) {
+        //This will be replaced with call to http
+        console.log("Username:"+this.username.value);
+        console.log("Email:"+this.email.value);
+        console.log("Password:"+this.password.value);
+        console.log("ConfirmPassword:"+this.passwordConfirm.value);
```
Comment (kpagacz) [inline frontend/src/pages/RegisterPage.vue:142]: Shall we replace this with the function? I think it's already merged, right?

Comment (karolkozlo) [reply frontend/src/pages/RegisterPage.vue:142]: Not in this task. I would like to separate component task and http layer task. We will implement whole registration in #47

File: frontend/src/components/common.less [hunk 2]
```diff
@@ -20,3 +20,13 @@
...
+
+
+//Some utilities
+
+.horizontal-line {
+    display: flex;
+    width: 100%;
+    height: 2px;
+    background-color: black;
+}
```
Comment (kpagacz) [inline frontend/src/components/common.less:32]:
```suggestion
}

```

File: frontend/src/pages/RegisterPage.vue [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+      } else {
+        this.passwordConfirm.errorMsg = '';
+      }
+    },
+    uploadAvatar() {
+      console.log("click on button");
+      this.$refs.fileInput.click();
+    },
+    changeAvatar(event) {
+      console.log("click on input");
```
Comment (kpagacz) [inline frontend/src/pages/RegisterPage.vue:124]: This can go, right?

File: frontend/src/pages/RegisterPage.vue [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+    setPasswordConfirm(newValue) {
+      this.passwordConfirm.value = newValue;
+      if(this.passwordConfirm.value != this.password.value) {
+        this.passwordConfirm.errorMsg = 'Passwords do not match'
+      } else {
+        this.passwordConfirm.errorMsg = '';
+      }
+    },
+    uploadAvatar() {
+      console.log("click on button");
```
Comment (kpagacz) [inline frontend/src/pages/RegisterPage.vue:120]: This can go, right?

File: frontend/src/pages/RegisterPage.vue [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+      color: @primary-dark-text-color;
+      font-size: 1.1rem;
+    }
+
+    .invisible-item {
+      display: flex;
+    }
+  }
+}
+</style>
```
Comment (kpagacz) [inline frontend/src/pages/RegisterPage.vue:221]:
```suggestion
</style>

```

File: frontend/src/pages/RegisterPage.vue [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+          <wof-input name="User name" :error="username.errorMsg" @change="setUsername" :maxLength="20"></wof-input>
+        </div>
+        <div class="register-page__input">
+          <wof-input name="Email address" :error="email.errorMsg" @change="setEmail"></wof-input>
+        </div>
+        <div class="register-page__input">
+          <wof-input name="Password" :type="'password'" :error="password.errorMsg" @change="setPassword"></wof-input>
+        </div>
+        <div class="register-page__input">
+          <wof-input name="Confirm Password" :type="'password'" :error="passwordConfirm.errorMsg" @change="setPasswordConfirm"></wof-input>
```
Comment (kpagacz) [inline frontend/src/pages/RegisterPage.vue:17]: Shall we stick to the lowercase after the first word?
```suggestion
          <wof-input name="Confirm password" :type="'password'" :error="passwordConfirm.errorMsg" @change="setPasswordConfirm"></wof-input>
```

File: frontend/src/pages/RegisterPage.vue [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+              <img :src="avatarImage" class="register-page__image" />
+              <div>
+                <input
+                  type="file"
+                  ref="fileInput"
+                  accept="image/png, image/jpeg"
+                  hidden
+                  @change="changeAvatar"/>
+                <wof-button type="button" icon="wof-camera" :size="1.2" @click="uploadAvatar">
+                  Upload Image
```
Comment (kpagacz) [inline frontend/src/pages/RegisterPage.vue:32]:
```suggestion
                  Upload image
```

File: frontend/src/pages/RegisterPage.vue [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+          </div>
+        </div>
+      </form>
+      <div class="horizontal-line"></div>
+      <div class="register-page__bottom">
+        <router-link to="/login" class="register-page__link">
+          Already have an account? Sign In
+        </router-link>
+        <wof-button :size="1.5" class="register-page__submit-button" @click="createAccount">
+          Create Account
```
Comment (kpagacz) [inline frontend/src/pages/RegisterPage.vue:45]:
```suggestion
          Create account
```

File: frontend/src/components/WofButton.vue [hunk 9]
```diff
@@ -10,7 +10,7 @@ export default {
   name: "WofButton",
 
   props: {
-    type: {
+    variant: {
```
Comment (kpagacz) [inline frontend/src/components/WofButton.vue:13]: Heh, my bad, I wasn't aware the type is already an attribute of the HTML button tag

**Review by kpagacz — COMMENTED** [review]
I tested it out, functionally it's alright. I have seen some minor clipping out of the div issues on a very small viewport:
image

Don't know if it can be helped or not. Otherwise looks good.

File: frontend/src/main.js [hunk 10]
```diff
@@ -3,18 +3,23 @@ import App from './App.vue'
...
-import WofButton from './components/WofButton.vue';
+import RegisterPage from './pages/RegisterPage.vue';
 const router = createRouter({
     history: createWebHistory(),
     routes: [
         { path: '/', component: WofHelloWorld },
         { path: '/login', redirect: '/' },
         { path: '/user/:userLogin', redirect: '/' },
+        { path: '/register', component: RegisterPage },
+        { path: '/login', redirect: '/' },
```
Comment (kpagacz) [inline frontend/src/main.js:16]: Duplicate of line 13, right?

File: frontend/src/pages/RegisterPage.vue [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+    }
+
+    .register-page__link {
+      color: @primary-dark-text-color;
+      font-size: 1.1rem;
+    }
+
+    .invisible-item {
+      display: flex;
+    }
```
Comment (kpagacz) [inline frontend/src/pages/RegisterPage.vue:216]: I cannot find this class used in the template. What am I missing?

File: frontend/src/pages/RegisterPage.vue [hunk 12]
```diff
@@ -0,0 +1,230 @@
...
+@media (max-width: 767.98px) {
+
+  .register-page__image-section {
+    align-items: center;
+
+    .register-page__image {
+      order: 2;
+    }
+  }
+}
```
Comment (kpagacz) [inline frontend/src/pages/RegisterPage.vue:229]: Nice! Although right now those guys are plastered together:
image

Introduce a gap and we are good to go?

File: frontend/src/pages/RegisterPage.vue (hunk 11 above)
Comment (karolkozlo) [reply frontend/src/pages/RegisterPage.vue:216]: Remains of the experiment. I will remove them

File: frontend/src/pages/RegisterPage.vue [hunk 13]
```diff
@@ -0,0 +1,227 @@
...
+
+@media (max-width: 767.98px) {
+
+  .register-page__image-section {
+    align-items: center;
+
+    .register-page__image {
+      margin-top: 0.5em;
```
Comment (kpagacz) [inline frontend/src/pages/RegisterPage.vue:222]: Please use the gap property, the flexbox does not have collapsing margins and margin makes it less intuitive and also harder to maintain.

See https://stackoverflow.com/a/20653092/8580062 for more info.

**Review by kpagacz — APPROVED** [review]

**Pull request merged**